store/views.py

```diff
@@ -12,58 +15,110 @@ def index(request):
 
 def bookDetailView(request, bid):
     template_name = 'store/book_detail.html'
-    context = {
-        'book': None, # set this to an instance of the required book
-        'num_available': None, # set this to the number of copies of the book available, or 0 if the book isn't available
-    }
-    # START YOUR CODE HERE
     
+    book = get_object_or_404(Book,id=bid)
+    num = BookCopy.objects.filter(book__title=book.title,status=True).count()
+
+    if(request.user.is_authenticated):
+        user = User.objects.get(username = request.user.username)
+        ratingofuser = rating.objects.filter(book = book,user = user).count()
+        if(ratingofuser>0):
+            rate = rating.objects.get(book = book,user = user).rate
+        else:
+            rate = 0.0
+    else:
+        rate = 0.0    
+    if num==0:
+        context = {
+            'book': book, # set this to an instance of the required book
+            'num_available': 'Sorry Book not available', # set this to the number of copies of the book available, or 0 if the book isn't available
+            'rate': rate,
+        }
+    else:
+        context = {
+            'book': book,
+            'num_available': num,
+            'rate': rate,
+        }
+    # START YOUR CODE HERE
     
     return render(request, template_name, context=context)
 
 
 @csrf_exempt
 def bookListView(request):
     template_name = 'store/book_list.html'
+
+    title = request.GET.get("title")
+    author = request.GET.get("author")
+    genre = request.GET.get("genre")
+    if(title and author and genre):
+        books = Book.objects.filter(title=title,author=author,genre=genre)
+    elif(title and author):
+        books = Book.objects.filter(title=title,author=author)
+    elif(title and genre):
+        books = Book.objects.filter(title=title,genre=genre)
+    elif(author and genre):
+        books = Book.objects.filter(author=author,genre=genre)
+    elif(title):
+        books = Book.objects.filter(title=title)
+    elif(author):
+        books = Book.objects.filter(author=author)
+    elif(genre):
+        books = Book.objects.filter(genre=genre)
+    else:
+        books = Book.objects.all()
```

Comment:
You could have done this in a simpler way.
Also, you should have used `i__contains` here instead of an exact match during search.